crates/serverless_downloader/src/lib.rs

```diff
@@ -20,11 +20,22 @@ pub fn get_bucket() -> Result<Bucket> {
         None,
     )?;
 
-    Ok(Bucket::new(
-        &bucket_name,
-        bucket_region.parse()?,
-        credentials,
-    )?)
+    let endpoint = env::var("S3_ENDPOINT");
+    let bucket = if endpoint.is_ok() {
+        Bucket::new(
+            &bucket_name,
+            Region::Custom {
+                region: bucket_region.parse()?,
+                endpoint: endpoint.unwrap(),
+            },
+            credentials,
+        )?
+        .with_path_style();
+    } else {
+        Bucket::new(&bucket_name, bucket_name.parse()?, credentials)?;
```

Comment:
@QuiiBz Maybe you cloud have a close look if this code is fine and nicely written as I am not a rust expert 🙈

Reply:
I've updated it a bit using a `match` statement, but your code was great!